### PR TITLE
IGNITE-15277 System views support

### PR DESCRIPTION
**Pull request by alex-plekhanov**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/SystemViewScan.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    private final Predicate<Row> filters;
+
+    /** */
+    private final Function<Row, Row> rowTransformer;
+
+    /** Participating colunms. */
+    private final ImmutableBitSet requiredColumns;
+
+    /** System view field names (for filtering). */
+    private final String[] filtrableFieldNames;
```
Comment (zstan) [inline modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/SystemViewScan.java:63]:
```suggestion
    private final String[] filterableFieldNames;
```

Comment (alex-plekhanov) [reply modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/SystemViewScan.java]: Fixed

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/schema/CacheColumnDescriptor.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    /** */
+    public boolean key();
+
+    /** */
+    public Object value(ExecutionContext<?> ectx, GridCacheContext<?, ?> cctx, CacheDataRow src)
+        throws IgniteCheckedException;
+
+    /** */
+    public void set(Object dst, Object val) throws IgniteCheckedException;
```
Comment (zstan) [inline modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/schema/CacheColumnDescriptor.java:39]: I suppose that minimal java docs are necessary for all of these methods.

**Review by zstan — APPROVED** [review]

**Comment by alex-plekhanov** [comment]
Merged to sql-calcite branch

**Pull request closed without merging**